jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EJBTimerScheduler.java

```diff
@@ -218,30 +181,16 @@ private boolean isSessionNotFound(Exception e) {
 
     @FunctionalInterface
     private interface Transaction<I> {
-
         void doWork(I item) throws Exception;
     }
 
-    private <I> void transaction(Transaction<I> operation, I item) throws Exception {
-        try {
-            utx.begin();
-            operation.doWork(item);
-            utx.commit();
-        } catch(RollbackException e) {
-            logger.warn("Transaction was rolled back for {} with status {}", item, utx.getStatus());
-            if(utx.getStatus() == javax.transaction.Status.STATUS_ACTIVE) {
-                utx.rollback();
-            }
-            throw new RuntimeException("jbpm timer has been rolledback", e);
-        } catch (Exception e) {
-            try {
-                utx.rollback();
-            } catch (Exception re) {
-                logger.error("transaction could not be rolled back", re);
-            }
-
-            throw e;
-        }
+    @TransactionAttribute(value = TransactionAttributeType.REQUIRES_NEW)
```

Comment:
REQUIRES_NEW seems redundant here. Doesn't the container already start a tx on timer start? To me it seems REQUIRED should be sufficient here.

Reply:
You are right about container creating a transaction already, however, here we want to split the "sunny day" path (the timer task execution) and the "rainy day" path (tha handling of any excepton launhed by the timer task) in two different transactions. It is a bit "heavyweight" but safer. However see change [here](https://github.com/kiegroup/jbpm/pull/2365/files#diff-5669039d6beade81bd11322820ff3c35dca83d5ada6986d268e1ef9879f06a65R192) to ensure rollback is happening.